sport/app/football/feed/agent.scala

```diff
@@ -154,10 +154,30 @@ class CompetitionAgent(
         }
       }
 
-      //it is important that newMatches are at the start of the list here
-      comp.copy(matches = (newMatches ++ comp.matches).sorted(MatchStatusOrdering).distinctBy(_.id).sortByDate)
+      val allMatches = (newMatches ++ comp.matches).sorted(MatchStatusOrdering).distinctBy(_.id).sortByDate
+
+      val updatedMatches = allMatches.filter(footballMatch =>
```

Comment:
scala thought - a `filterNot` might make this a little more readable.

```scala
allMatches.filterNot(footBallMatch => {
  val isPlaceholder = placeholderIndicators.exists(footBallMatch.homeTeam.name.contains)
  val hasMatchWithMatchingTime = allMatches.exists(_.date == footBallMatch.date)
  
  isPlaceholder && hasMatchWithMatchingTime
})
```

Reply:
This looks a lot better indeed but we get a failing test. I think it should be written like this

```
      val updatedMatches = allMatches.filterNot(footballMatch => {
        val isPlaceHolder = isPlaceholderMatch(footballMatch)
        val nonPlaceHolderMatchWithSameDateExists = allMatches.exists(existingMatch =>
          footballMatch.date == existingMatch.date && !isPlaceholderMatch(existingMatch),
        )

        isPlaceHolder && nonPlaceHolderMatchWithSameDateExists
      })
```

which I'm not sure it's very readable either. For now I'm going to merge so that we avoid seeing the issue tonight. I will try to improve this tomorrow!